src/passes/OptimizeInstructions.cpp

```diff
@@ -990,17 +952,24 @@ struct OptimizeInstructions
           }
         }
       }
-      if (binary->op == OrInt32) {
+      if (binary->op == OrInt32 || binary->op == XorInt32) {
         // an or flowing into a boolean context can consider each input as
         // boolean
         binary->left = optimizeBoolean(binary->left);
         binary->right = optimizeBoolean(binary->right);
-      } else if (binary->op == NeInt32) {
-        // x != 0 is just x if it's used as a bool
+      }
```

Comment:
why remove this `else`? it can be less efficient your way, as the later if is always checked, but it doesn't need to be if we entered the first if.